### ISSUE TITLE
CLI0143E Invalid precision value SQLSTATE=HY104, or Numeric Value out of range - passing Decimal value to db2

### ISSUE THREAD

**Issue by lakhanpalv** [ibmdb/node-ibm_db#636]
Would like to pass a Decimal value to db2 using Stored Proc, Decimal (15,2).
Here is how my param object looks like:
{ParamType: 'INPUT', CType: 8, DataType: 3, Data: 34534411.3, Length:15}

Getting Error 
CLI0143E Invalid precision value SQLSTATE=HY104

If I use {ParamType: 'INPUT', DataType:'DECIMAL(15,2)', Data: 232323233.23}
Then i get error Numeric value out of range.

Any suggestion?
What's the best way to pass Decimal values above 8 numbers in length to a SP, as I noticed if I pass numbers/decimals below 8 numbers it works fine, with 2nd approach, without specifying the length or precision.

Thanks.
Vivek

**Comment by bimalkjha** [comment]
@lakhanpalv {ParamType: 'INPUT', CType: 8, DataType: 3, Data: 34534411.3, Length:15} should work not the  DataType:'DECIMAL(15,2)'. If you can share some test program to reproduce the problem, it'll help to find the root cause faster. Thanks.

**Comment by lakhanpalv** [comment]
Thanks Bimal,
So I tried your suggested solution, Here is the payload which is working with one way but not the other way, object payload i am trying to pass to SP
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 345111.32, Length: 15 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 819111.22,Length: 15 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 3434411.3,Length: 15 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 3, Length: 5 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 12, Length: 6 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 1, Length: 5 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 45, Length: 15 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 568,Length: 15 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 543,Length: 15 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 34343,Length: 15 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 87654321,Length: 15 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 23333.23,Length: 15 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 45, Length: 5 }
key  { ParamType: 'INPUT', CType: 8, DataType: 3, Data: 5432,Length: 15 }

Error during statement execute block: statement,: Error: [IBM][CLI Driver] CLI0143E  Invalid precision value. SQLSTATE=HY104
i.e. it didn't even reach db2 at this stage
"code block where i am creating statement is:"
```
stmt.execute(this.getParams(payload), (errStmt: Error, result: any, outparams: any) => {
    if (errStmt) {
        rejects(new Error(`error in create statement: ${errStmt}`));
    }
```


For same dataset if i use the below format, it works fine. but then i can't pass more then 7 or 8 character long number, Some of the fields can have decimal precision like 6,4 or some can be like 5,0, but i am not sure it's validating that, at statement creation time, is it aware of what db2 is expecting at this time?

same values worked data set If i pass them as DECIMAL
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 345111.32 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 819111.22 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 3434411.3 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 3 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 12 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 1 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 45 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 568 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 543 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 34343 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 87654321 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 23333.23 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 45 }
key  { ParamType: 'INPUT', DataType: 'DECIMAL', Data: 5432 }

Appreciate your help in looking into it.

Thanks,
Vivek

**Comment by lakhanpalv** [comment]
Hi Bimal,
I figured out the issue, I can pass the decimal values using my method i.e. { ParamType: 'INPUT', DataType: 'DECIMAL', Data: '5432233434.34' }, the only thing I found fixes it is that if Data is passed with single quotes instead of doing parseFloat(number). It worked fine for all kind of decimal numbers with any precision.  
I am closing the ticket. Thanks for looking into the issue :)
Vivek.
